design/backup-repo-config.md

```diff
@@ -0,0 +1,126 @@
+Therefore, we create a BackupRepository configMap as a template of the configurations to be applied to the backup repository CR when it is created by the BackupRepository controller. For an existing BackupRepository CR, the configMap is never visited, if users want to modify the configuration value, they should directly edit the BackupRepository CR.  
+
+The BackupRepository configMap is created by users in velero installation namespace. If the configMap is not there, nothing is specified to the backup repository CR, so the Unified Repo modules use the hard-coded values to configure the backup repository.  
+
+The BackupRepository configMap is backup repository type specific, that is, for each type of backup repository, there is one configMap. Therefore, a ```backup-repository-config``` label should be applied to the configMap with the value of the repository's type. During the backup repository creation, the configMap is searched by the label against the repository type, if no configMap is found, nothing is set to the backup repository CR.  
+
+### Configurations
+
+With the above mechanisms, any kind of configuration could be added. Here list the configurations defined at present:  
+```cacheLimitMB```: specify the size limit(in MB) for the local data cache. The more data is cached locally, the less data may be downloaded from the backup storage, so the better performance may be achieved. Practically, users can specify any size that is smaller than the free space so that the disk space won't run out. This parameter is for each repository connection, that is, users could change it before connecting to the repository. If a backup repository doesn't use local cache, this parameter will be ignored. For Kopia repository, this parameter is supported.  
+```enableCompression```: specify to enable/disable compression for a backup repsotiory. Most of the backup repositories support the data compression feature, if it is not supported by a backup repository, this parameter is ignored. Most of the backup repositories support to dynamically enable/disable compression, so this parameter is defined to be used whenever creating a write connection to the backup repository, if the dynamically changing is not supported, this parameter will be hornored only when initializing the backup repository. For Kopia repository, this parameter is supported and can be dynamically modified.  
+
+Below is an example of the BackupRepository configMap with the configurations:  
+json format:  
+```json
+{
+    "repoConfig": {
+       "cacheLimitMB": 2048,
+       "enableCompression": true      
+    }
+}
+```
+yaml format:  
+```yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: <config-name>
+  namespace: velero
+data:
+  repoConfig: |
+    {
+      "cacheLimitMB": 2048,
+      "enableCompression": true
```

Comment:
I also think it's worth considering allowing other repo configuration options to be configurable via this design. Namely thinking of different kopia repo options e.g.:
https://github.com/wawa0210/velero/blob/89c10ddcc04e4dc7354f284434c2ddb8c121780c/pkg/repository/udmrepo/repo_options.go#L63-L65

Reply:
@weshayutin ^^

Reply:
Let's do this incrementally in future based on the requirements.
For now, the 3 options you mentioned are too advanced(especially for the hash algorithm and chunking algorithm, without a good knowledge of the deduplication and the status of their own data, users cannot decide a better value than the default ones) and not general enough in terms of unified repository concepts(many backup repositories/storages are using private chunking/hashing algorithms and don't allow users to change, because those have heavy impacts on the QOS of the repo).